backend/controller/dal/dal.go

```diff
@@ -180,12 +185,39 @@ func New(ctx context.Context, conn libdal.Connection, encryption *encryption.Ser
 	return d
 }
 
+func (d *artefactDAL) GetArtefactDigests(ctx context.Context, digests [][]byte) ([]artefacts.ArtefactRow, error) {
+	results, err := d.db.GetArtefactDigests(ctx, digests)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve artefact digests: %w", err)
+	}
+	return slices.Map(results, func(in dalsql.GetArtefactDigestsRow) artefacts.ArtefactRow {
+		return artefacts.ArtefactRow{ID: in.ID, Digest: in.Digest}
+	}), nil
+}
+
+func (d *artefactDAL) CreateArtefact(ctx context.Context, digest []byte, content []byte) (int64, error) {
+	id, err := d.db.CreateArtefact(ctx, digest, content)
+	if err != nil {
+		return id, fmt.Errorf("could not create artefact: %w", err)
+	}
+	return id, nil
+}
+
+func (d *artefactDAL) GetArtefactContentRange(ctx context.Context, start int32, count int32, iD int64) ([]byte, error) {
+	bytes, err := d.db.GetArtefactContentRange(ctx, start, count, iD)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve artefact content range: %w", err)
+	}
+	return bytes, nil
+}
+
 type DAL struct {
 	*libdal.Handle[DAL]
 	db dalsql.Querier
 
 	leaseDAL   *leasedal.DAL
 	encryption *encryption.Service
+	Registry   artefacts.Registry
```

Comment:
This is okay for now, but this should be moved out of the DAL at some point.

Reply:
encryption exposes a server, how does applying that scheme to the artefacts registry sound?

Reply:
Yep perfect. That's the pattern we're moving to everywhere, so that would be ideal.